modules/objc/generator/CMakeLists.txt

```diff
@@ -83,4 +83,10 @@ add_custom_target(gen_opencv_objc_source DEPENDS ${objc_generated_files}
             "${CMAKE_CURRENT_BINARY_DIR}/gen_objc.json"
 )
 
-add_dependencies(opencv_world gen_opencv_objc_source)
+if(BUILD_opencv_world)
+  add_dependencies(opencv_world gen_opencv_objc_source)
+else()
+  foreach(m ${OPENCV_OBJC_MODULES})
+    add_dependencies(${m} gen_opencv_objc_source)
```

Comment:
Are other modules depend on generated code?

Did you try?
```
add_dependencies(${the_module} gen_opencv_objc_source)
```

Reply:
OK - I've figured it out now. All I needed to do was add the target **gen_opencv_objc_source** to the ALL target